index/index.go

```diff
@@ -893,7 +928,8 @@ func (r *Reader) Series(id uint64, lbls *labels.Labels, chks *[]chunks.Meta) err
 }
 
 // Postings returns a postings list for the given label pair.
-func (r *Reader) Postings(name, value string) (Postings, error) {
+// 'reusePos' is used if it is a 'bigEndianPostings'.
```

Comment:
This feels like a layering violation

Reply:
Did not understand what is being violated.

Reply:
You're altering a generic API merely for a micro-optimisation in one of its implementations.

Reply:
This saved like 10% of allocs, so I would not count it as micro-opt. But I agree that altering this generic API is not clean. I will see if I can do it in a better way, else might as well leave it as allocs got reduced in other places to cover it up.

Reply:
Initially I was thinking to have `Reset(...)` method for `Postings` interface, but could not decide what would be the type of the argument. Now I see the argument can be `interface{}` and particular `Postings` interface implementations will typecast to needed types and reset and return `true` if reset is successful. This way we will use the generic `Reset(...) bool` function without going to specifics.

If `Reset(...)` returned `false`, it means that we need to create another object.

Does this look sane in any way?

Reply:
That sounds okay, but I'm not sure why you need the bool return.

Reply:
Some of the postings have `[]byte` as base (example `bigEndianPostings`) and some have `[]uint64` (example `ListPostings`). So if I pass `[]uint64` to `bigEndianPostings.Reset`, I am not sure if the `Reset` would make sense, so in that case I would return false.

Reply:
I have added that method in the `Postings` interface.